sycl/include/sycl/ext/oneapi/matrix/static-query.hpp

```diff
@@ -305,6 +358,42 @@ struct tpu_params<tpu::dpas, void, void, void, M, N, K> {
       combinations[i + ii].nsize = 8;
     }
     i = 4;
+    combinations[i].atype = matrix_type::sint8;
+    combinations[i].btype = matrix_type::uint8;
+    combinations[i].ctype = matrix_type::sint32;
```

Comment:
This pattern is used in several places, but it looks like we're only assigning `atype`, `btype`, and `ctype` for the `i`'th combination and not `i+1`, `i+2`, and `i+3`.  Can you please verify that all of the necessary types are assigned correctly?

One way to do this is to follow the suggestion above and initialize the `combinations` array when it is declared vs. here in the constructor.